lib/upload.js

```diff
@@ -276,13 +284,23 @@ class Upload {
 
     cb = cb || function () {};
     if (shouldTerminate && this.url != null) {
-      Upload.terminate(this.url, this.options, (err, xhr) => {
-        if (err) {
-          return cb(err, xhr);
-        }
+      const terminate = () => {
+        Upload.terminate(this.url, this.options, (err, xhr) => {
+          if (err) {
+            if (this._shouldRetry(err)) {
+              const delay = this.options.retryDelays[this._retryAttempt++];
```

Comment:
You should `this._retryAttempt = 0;` before starting the retries for the termination.

Reply:
oh ok, I wasn't sure if it was desired or not